velox/functions/sparksql/tests/DateTimeFunctionsTest.cpp

```diff
@@ -142,5 +142,51 @@ TEST_F(DateTimeFunctionsTest, toUnixTimestamp) {
   EXPECT_THROW(evaluateOnce<int64_t>("to_unix_timestamp()"), VeloxUserError);
 }
 
+TEST_F(DateTimeFunctionsTest, lastDay) {
+  const auto lastDay = [&](std::optional<Date> date) {
+    return evaluateOnce<Date>("last_day(c0)", date);
+  };
+  Date d0;
+  parseTo("2015-02-28", d0);
+  Date d1;
+  parseTo("2015-02-28", d1);
```

Comment:
would it make since to avoid replication here by creating a helper function

auto test =[&](const auto& inputDate, const auto& expectedDate){
    Date d0;
    Date d1;
    parseTo(inputDate, d0);
    parseTo(expectedDate, d1);
    EXPECT_EQ(lastDay(d1), d0);
};
test("2015-03-31","2015-03-27");
test("2015-03-31","2015-03-27");
test("2015-03-31","2015-03-27");
...etc

Reply:
@laithsakka done